### DISCUSSION TITLE
I have a question about the 'output_frequency'

### DISCUSSION THREAD

**Discussion by YiyingWang-00** [WAM2layers/WAM2layers#218] [Q&A] [answered]
Hello! I have a question about the output_frequecy'

**Comment by YiyingWang-00** [comment]
When the output_frequecy' is 3h, the result is just one file in a day for example, '2003-12-01_s_track.nc', but not per 3h for a result. I think the new results in next 3h covered the last one. I tried to modify the backtrack.py, in line 38-40 by adding %H:
def output_path(date, config):
    output_dir = config.output_folder
    return f"{output_dir}/{date.strftime('%Y-%m-%d-%H')}_s_track.nc"

I don't know whether is true. Can you help me?
![1683811710852](https://github.com/WAM2layers/WAM2layers/assets/112865233/a0484f3a-6a02-481b-9152-5cc387fdb31f)

**Reply by YiyingWang-00** [reply]
I want to get tracking results every three hours

**Reply by Peter9192** [reply]
You're right, this is a bug. I also encountered it in #201 and added a similar fix by changing this line to include the time stamp:

https://github.com/WAM2layers/WAM2layers/blob/32cfdd57b75466e873532b86ff294fee7f8c0b34/src/wam2layers/tracking/backtrack.py#L39

It would be better to make a separate pull request for this bugfix - and maybe check it a bit more thoroughly. Would you like to try and open a pull request? See the instructions [here](https://wam2layers.readthedocs.io/en/latest/develop.html#collaborative-workflow)

**Reply by YiyingWang-00** [reply]
Thanks for replying! Of course, I would like to open a pull request.

**Reply by ruudvdent** [reply]
@Peter9192 , should we convert this to an issue?

**Reply by Peter9192** [reply]
Makes sense

**Reply by YiyingWang-00** [reply]
How can I use WAM2layer to experiment the forward tracking? @Peter9192

**Reply by Peter9192** [reply]
We need to add back forward tracking. It is similar to back tracking but just the other way, so to put it very simply, you copy the backtrack.py script, change the direction of the time loops and some pluses and minuses. But in the real world it's a bit more involved than that. @ruudvdent is planning to work on this I believe sometime this summer.

**Reply by ruudvdent** [reply]
correct, we have listed it as issue #8

**Reply by YiyingWang-00** [reply]
Ok. Thanks! Looking forward to further progress!

**Reply by Peter9192** [reply]
#231 properly fixed the output filenames to include hours and minutes.

**Comment by YiyingWang-00** [comment]
If I want to set the output_frequency to 30d, does it work? But I have a question, why the date is 12-19? 
![1683812883068](https://github.com/WAM2layers/WAM2layers/assets/112865233/ae0a5235-1a91-4c11-8eb2-8b0cdbb30f9d)

![1683812912781](https://github.com/WAM2layers/WAM2layers/assets/112865233/c8b32a3a-b4c6-4b27-8788-1f5c80f310a0)

**Reply by Peter9192** [reply]
The code that determines when the state is written to disk is here:

https://github.com/WAM2layers/WAM2layers/blob/a877e18b0f8c669cdef83edce894ed4f2b0a63a6/src/wam2layers/tracking/backtrack.py#L508C32-L511

As you can see, it uses pandas' [`floor`](https://pandas.pydata.org/docs/reference/api/pandas.Timestamp.floor.html) function for timestamps. You can see for yourself with:

```py
import pandas as pd
for date in pd.date_range('20001201', '20001231'):
    print(date.floor('30d'))
```

Understanding why it behaves like this is one thing. Whether that behaviour makes sense is a whole different question... If you have ideas about what behaviour would be better, and how we could implement that, your input is very welcome 😃

**Reply by YiyingWang-00** [reply]
Yes. Actually, I want to do some research on a monthly scale, and I wonder whether I can input the mean monthly data directly to calculate the mean monthly results or input the hourly data to output the monthly results.

**Reply by Peter9192** [reply, marked as answer]
I think monthly data is probably too coarse to get realistic estimates of moisture transport. What do you think, @ruudvdent?

**Reply by ruudvdent** [reply]
I agree, that would not make much sense.
